src/app/skill.service.spec.ts

```diff
@@ -1,12 +1,81 @@
+import { SkillService } from 'src/app/skill.service';
 import { TestBed } from '@angular/core/testing';
-
-import { SkillService } from './skill.service';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { of } from 'rxjs/internal/observable/of';
+// import { SkillService } from './skill.service';
 
 describe('SkillService', () => {
-  beforeEach(() => TestBed.configureTestingModule({}));
+  let skillService: SkillService; //added for testing
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+    imports: [HttpClientTestingModule],
+    providers: [SkillService]
+  });
+  skillService = TestBed.get(SkillService);//added for testing 
+});
 
   it('should be created', () => {
-    const service: SkillService = TestBed.get(SkillService);
-    expect(service).toBeTruthy();
+    // const service: SkillService = TestBed.get(SkillService);
+    expect(skillService).toBeTruthy();
+  });
+  describe('retrieveAllSkills', () => {
+    it('should return all skills', () => {
+      const skillResponse = [
+        {
+          id: 1,
+          title: 'Java'
+        },
+        {
+          id:2,
+          title:'CSS'
+        },
+        {
+          id: 3,
+          title: 'JavaScript'
+        }
+      ];
+      let response;
```

Comment:
Again, this series of statements looks like it is not actually testing anything.